File: SiriusJavaClient-Win32/src/main/java/org/sirius/client/win32/classes/MainWindow.java
/**
 * .
 */
package org.sirius.client.win32.classes;

import org.sirius.client.win32.Win32Client;
import org.sirius.client.win32.types.Win32Locator;

/**
 * @author Myk Kolisnyk
 * .
 */
public class MainWindow extends TopLevelWindow {

    /**
     * .
     * @param client .
     * @param locator .
     */
    public MainWindow(final Win32Client client, final Win32Locator locator) {
        super(client, locator);
    }

    /**
     * .
     * @param executable .
     * @param params .
     * @param workingDir .
     * @throws Exception .
     */
    public final void start(final String executable, final String params,
            final String workingDir) throws Exception {
        client.core().window()
                .start(locator.getHwnd(), executable, params, workingDir);
    }
}

File: sirius-ruby-client-win32/lib/sirius/win32/classes/MainWindow.rb
module Sirius
  module Client
    module Win32
      module Classes
        class MainWindow < TopLevelWindow
          def initialize(client=nil, locator=nil)
            super(client, nil, locator)
          end

          def start(executable, params, workingDir)
            @client.core.window.start(@locator.hwnd, executable, params, workingDir)
          end
        end
      end
    end
  end
end